doc/aaa/TACACS+ Design.md

```diff
@@ -0,0 +1,521 @@
+|                                |       |                     |
++---------------+----------------+       +---------------------+
+```
+
+
+ - The hostcfg enforcer reads data from configDB to configure host environment.
+   - The AAA config module in hostcfg enforcer is responsible for modifying Auditd configuration files in host.
+   - For how TACACS+ config file update, please check [TACACS+ Authentication](#TACPLUS-Authentication)
+
+The following figure show how Auditd config an TACACS+ config update by ConfigDB.
+```
++--------------------------------+       +---------------------+
+| Auditd                         |       |                     |
+|   +-------------------------+  |       |  +--------------+   |
+|   | Auditd config file      <-------------+ Accounting   |
+|   |                         |  |       |  |    Config    |   |
+|   +-------------------------+  |       |  +--------------+   |
+|                                |       |                     |
+|   +-------------------------+  |       |  +--------------+   |
+|   | TACACS+ config file     <-------------+  AAA Config  |   |
+|   +-------------------------+  |       |  +--------------+   |
+|                                |       |                     |
+|                                |       |  HostCfg Enforcer   |
++--------------------------------+       +----------^----------+
+                                                    |
+           +---------+                      +-------+--------+
+           |         |                      |                |
+           |   CLI   +---------------------->    ConifgDB    |
+           |         |                      |                |
+           +---------+                      +----------------+
+```
+
+
+## 4.4 ConfigDB Schema
+ - TACACS+ Authorization and Accounting will use existing tables
+   - AAA Table.
+   - TACPLUS Table
+   - TACPLUS_SERVER Table.
+
+ - For more detail of existing tables, please check [TACACS+ Authentication](#TACPLUS-Authentication)
+
+## 4.5 CLI
+ - The existing TACACS+ server config command will not change.
+ - Add following command to enable/disable TACACS+ authorizarion.
+```
+    // authorization with TACACS+ server and local
+    config aaa authorization tacacs local
+    
+    // authorization with TACACS+ server
+    config aaa authorization tacacs
+    
+    // authorization with TACACS+ local
+    config aaa authorization local
+```
+
+ - Add following command to enable/disable TACACS+ accounting.
+```
+    // accounting with TACACS+ server and local syslog
+    config aaa accounting tacacs local
+
+    // accounting with TACACS+ server
+    config aaa accounting tacacs
+
+    // accounting with local syslog
+    config aaa accounting local
+```
+
+ - When config AAA authorization with "no" prefix, SONiC will use local authorization, so following commands have same effect
+```
+    no config aaa authorization tacacs local
```

Comment:
>no config [](http://example.com/codeflow?start=4&length=9)

It is difficult to implement `no config`, can we just config it as `local` as the last resort?

Reply:
'no config' have same effect with local.
I will update document.